### ISSUE TITLE
WandB dropping items when logging LR or val_loss with accumulate_grad_batches > 1

### ISSUE THREAD

**Issue by tadejsv** [PyTorchLightning/pytorch-lightning#5469]
## 🐛 Bug

As you can see in the [BoringModel](https://colab.research.google.com/drive/1w0lbH4UdMVPfvxUJjHgBXRWZrxyIcpGG?usp=sharing), I get the following warnings from WandB logger:

```
wandb: WARNING Step must only increase in log calls.  Step 49 < 98; dropping {'lr-SGD': 0.1}.
wandb: WARNING Step must only increase in log calls.  Step 99 < 198; dropping {'lr-SGD': 0.1}.
wandb: WARNING Step must only increase in log calls.  Step 149 < 199; dropping {'lr-SGD': 0.1}.
wandb: WARNING Step must only increase in log calls.  Step 149 < 298; dropping {'lr-SGD': 0.1}.
wandb: WARNING Step must only increase in log calls.  Step 156 < 299; dropping {'val_loss': 3.9880808209860966e-14, 'epoch': 0}.
```

This occurs when I add the following to the basic BordigModel:
- log train loss with `self.log()`
- add WandB logger
- add `accumulate_grad_batches > 1`
- either add `LearningRateMonitor` callback or log validation loss with `self.log()` (or both, as in the colab)

If any of these things is removed, the error doesn't occur. 

The end result is that the LR metrics are note being logged at all. Worse than that, validation loss (and any other metrics that there would be!) do not get logged, making the logger useless.

**Comment by tadejsv** [comment]
Also if it helps: this occurs in every version down to (and including) 1.0.9, but does not occur in 1.0.8

**Comment by borisdayma** [comment]
Do you have any call to `wandb.log()` in your validation loop. If so, you can try to add `commit=False`.
It seems that for some reason the wrong step is being used in the validation loop. However `self.log()` is supposed to handle correctly the right step (this could be an issue with gradient accumulation too).

You will be able to bypass these issues with #5194 by using `sync_step=False`.

**Comment by siddk** [comment]
Any update on when #5194 will be merged in (sync_step=False functionality).

Additionally, I'm not sure that `sync_step=False` will entirely solve the gradient accumulation problem. Specifically, [this line](https://github.com/PyTorchLightning/pytorch-lightning/blob/c71c0fb2bddf175c9b9caa553603aa65813a651e/pytorch_lightning/loggers/wandb.py#L176) just means that the Logger will try logging at `trainer.global_step` which is still out of sync with validation?

For example, here's an example use-case. I set `max_steps=100` in my trainer, and `val_check_interval=50` (validates every 50 steps). Gradient Accumulation is set with `accumulate_grad_batches=2`.

`global_step` for my Trainer will reflect a range from 0-200, while for validation, it'll show (50, 100), even though it should map directly to (100, 200).

Is there a way to fix this more cleanly? Pinging @SeanNaren because I've discussed this problem with him via the Lightning Slack.

**Comment by tadej-redstone** [comment]
Any update on this?

**Comment by borisdayma** [comment]
The wandb workaround has been merged #5194 and will be available in PL v1.2.

**Comment by tadej-redstone** [comment]
Thanks, can confirm this is the case in the BoringModel notebook

**Comment by borisdayma** [comment]
This was fixed with https://github.com/PyTorchLightning/pytorch-lightning/pull/5931.
You can now just log at any time with `self.log('my_metric', my_value)` and you won't have any dropped value.
Just choose your x-axis appropriately in the UI (whether `global_step` or just the auto-incremented `step`).
